### ISSUE TITLE
geo bounds aggregation: Add support for geo_shape.

### ISSUE THREAD

**Issue by yeroc** [elastic/elasticsearch#7574]
Some time ago I created an enhancement request to support a bounding box aggregation (#5634) as this is very useful for mapping applications to allow a map to zoom to the extent (bounding box) of all matches.  This was implemented in v1.3.0 but unfortunately only for the geo_point type.

**Comment by yeroc** [comment]
Any update on this?  Does this require some additional features in Lucene or does this require changes to ElasticSearch only?

**Comment by nknize** [comment]
No additional Lucene features required to support this feature.  I'm currently refactoring some legacy ```geo_point``` code to reuse ```geo_shape``` logic that fixes polygonal ambiguity issues.  Once that's done adding support for this request will be relatively straight-forward.

**Comment by yeroc** [comment]
@nknize Thanks a lot for the update.

**Comment by emmanuelmathot** [comment]
Great

**Comment by emmanuelmathot** [comment]
Any planned date for release?

**Comment by clintongormley** [comment]
@nknize just assigned this to you so you don't miss it

**Comment by clintongormley** [comment]
@nknize just pinging for an update on this.

**Comment by carlosvega** [comment]
+1

**Comment by nknize** [comment]
2.2 for sure and can be back ported to 2.0+ (@clintongormley thoughts on back porting?).

**Comment by clintongormley** [comment]
@nknize new features/enhancements go into the current stable branch and above only (ie 2.2+ at the moment)

**Comment by samchorlton** [comment]
What is the latest on this?

**Comment by nknize** [comment]
This was put on hold to prioritize spatial indexing performance issues in lucene 6.0. Once we get that stable this will move back up in the list.

**Comment by samchorlton** [comment]
Ok cool thanks for letting me know. Do you know if this will add support to allow aggregation of geoshape polygons or is this just for aggregation of points by polygons?

**Comment by nknize** [comment]
Yes, its to provide `geo_bounds` support for `geo_shape` field types. So `geo_shape` fields will be aggregated into a single bounding box.

**Comment by omarps** [comment]
+1

**Comment by yeroc** [comment]
@nknize I see the Lucene 6.1 release notes mention improvements to spatial indexing so assume the work you mention in your previous notes has been completed.  Does this mean this ticket is on deck now or is there still more work to be done prior?  Thanks!

**Comment by oleg1628** [comment]
+1

**Comment by howlowck** [comment]
How are we doing on this feature?

**Comment by aparnavarma92** [comment]
+1 this would be very useful functionality

**Comment by yeroc** [comment]
@nknize Do you have an update on this?

**Comment by hanoch** [comment]
👍

**Comment by rtrujill007** [comment]
👍 @nknize @sherry-ger Any updates on this issue?

**Comment by emmanuelmathot** [comment]
Is there any technical difficulty hard to solve for this issue?

**Comment by stevewillard** [comment]
What ever happened with this issue? Is there a branch where the work for this was started?

**Comment by hanoch** [comment]
@sherry-ger @nknize - any updates on this issue?

**Comment by clintongormley** [comment]
We won't be working on this until BKD-backed geoshapes are merged

**Comment by hanoch** [comment]
@clintongormley when is the "BKD-backed geoshapes" planned to be merged?

**Comment by clintongormley** [comment]
@hanoch when it is ready.  I can't be any more specific than that

**Comment by bilalvdemir** [comment]
Hi, this feature has complate?

**Comment by hanoch** [comment]
any progress on that?

**Comment by refaelasraf** [comment]
Have any updates on that ?

**Comment by hanoch** [comment]
@nknize
is this now addressed with the ES 6.5.0 release and https://github.com/elastic/elasticsearch/pull/32730 PR merge?

**Comment by imotov** [comment]
@hanoch no, we still need get #32039 in first

**Comment by nknize** [comment]
We also won't be able to add `geo_shape` aggregations until we come up w/ a reasonable doc value representation for `geo_shape`. WKB seems the best option at the moment but we need to investigate docValue size limits in lucene. 

It is possible that we will need to limit the number of shape vertices to comply w/ Lucene docValue size limits for `geo_shape` fields that have docValues are enabled. /brainstorm /cc @jpountz

**Comment by imotov** [comment]
Depends on #37206

**Comment by babadofar** [comment]
Apparently Kibana believes this has been fixed already. See this bug report
https://github.com/elastic/kibana/issues/31589

**Comment by thomasneirynck** [comment]
This is a requirement to be able to do layer-fitting on layers backed by `geo_shape` fields in the maps app. cf. https://github.com/elastic/kibana/issues/33509

**Comment by talevy** [comment]
@hanoch there is some progress. there is a feature branch that this issue is tracking: https://github.com/elastic/elasticsearch/issues/37206.

The feature branch has partial support for aggregations for now, one being `geo_bounds`: https://github.com/elastic/elasticsearch/pull/45413

Unfortunately, I cannot share an ETA for this work, but I can say that it is actively being worked on!

**Comment by talevy** [comment]
this was implemented in https://github.com/elastic/elasticsearch/pull/55328!
